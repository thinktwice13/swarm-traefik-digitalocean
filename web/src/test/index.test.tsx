import Home from '@pages/index';
import { render, screen } from '@testing-library/react';

describe('Sample test', () => {
  it('Passes', async () => {
    (global as any).fetch = jest
      .fn()
      .mockResolvedValueOnce({
        json: () =>
          Promise.resolve({
            data: [
              { id: 7654567, passphrase: 'kjuyhgy' },
              { id: 8765678, passphrase: 'kiuy6545tgh' },
            ],
          }),
      })
      .mockResolvedValueOnce({ json: () => Promise.resolve({ data: null }) });

    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
    expect(await screen.findByRole('page-title')).toHaveTextContent(
      'Swarm Traefik DigitalOcean'
    );
  });
});
