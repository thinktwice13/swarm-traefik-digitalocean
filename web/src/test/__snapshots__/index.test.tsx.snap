// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample test Passes 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1b2pnul"
  >
    <div
      class="chakra-stack css-d49qdf"
    >
      <h1
        class="chakra-heading css-0"
        role="page-title"
      >
        Swarm Traefik DigitalOcean
      </h1>
      <div
        class="chakra-stack css-owjkmg"
      >
        <p
          class="chakra-text css-1pkaall"
        >
          Docker Swarm microservices deployed on DigitalOcean
        </p>
        <div
          class="css-1hq7shv"
        >
          <span
            class="css-1ehzxp"
          >
            nodejs
          </span>
          <span
            class="css-1ehzxp"
          >
            nextjs
          </span>
          <span
            class="css-1ehzxp"
          >
            typescript
          </span>
          <span
            class="css-1ehzxp"
          >
            https
          </span>
          <span
            class="css-1ehzxp"
          >
            auth
          </span>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-owjkmg"
    >
      <form>
        <div
          class="chakra-stack css-84zodg"
        >
          <input
            class="chakra-input css-0"
            placeholder="Login/Register"
            type="text"
            value=""
          />
          <button
            class="chakra-button css-5c2176"
            type="submit"
          >
            Log In
          </button>
        </div>
      </form>
      <div
        class="css-b8u46k"
      />
    </div>
  </div>
</DocumentFragment>
`;
